bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableConfiguration.java

```diff
@@ -81,6 +84,25 @@
     return CONNECTION_CLASS;
   }
 
+  /**
+   * Set up connection impl classes. If org.apache.hadoop.hbase.client.AsyncConnection exists, set
+   * up async connection impl class as well.
+   */
+  private static Configuration configureConnectionClass(Configuration configuration) {
+    configuration.set(HBASE_CLIENT_CONNECTION_IMPL, getConnectionClass().getCanonicalName());
+    try {
+      // Set up HBase async registry impl if AsyncConnection class exists
+      Class.forName(HBASE_ASYNC_CONNECTION_CLASS);
```

Comment:
I think you should check for the classes that you actually want to use (BIGTABLE_HBASE_CLIENT_ASYNC_CONNECTION_CLASS & BIGTABLE_HBASE_CLIENT_ASYNC_REGISTRY_CLASS) instead of which hbase classes exist.

Or maybe check for all 3?

We can end up in a situation where the application has a mismatch of hbase and bigtable-hbase versions. ie the classpath can look like: hbase-client 1.x, bigtable-hbase-1x, hbase-client 2.x.
Since the classpath is searched in order, this might have worked before this change, but now it will crash.